Products/CMFPlone/tests/test_recyclebin.py

```diff
@@ -89,6 +89,50 @@ def test_recyclebin_settings(self):
         self.assertEqual(settings.retention_period, 30)
         self.assertEqual(settings.maximum_size, 100)
 
+    def test_recyclebin_permission(self):
+        """Test permission checks for the recycle bin"""
+        # As Manager role, should have access
+        self.assertTrue(self.recyclebin.check_permission())
+
+        # Create a test user with limited permissions
+        from AccessControl import getSecurityManager
+        from AccessControl.SecurityManagement import newSecurityManager
+        from AccessControl.SecurityManagement import setSecurityManager
+
+        # Store the current security manager
+        old_security_manager = getSecurityManager()
+
+        try:
+            # Create a regular Member user
+            self.portal.acl_users._doAddUser("testuser", "password", ["Member"], [])
+
+            # Log in as the test user
+            user = self.portal.acl_users.getUser("testuser")
+            user = user.__of__(self.portal.acl_users)
+            newSecurityManager(None, user)
```

Comment:
Use the login function from plone.app.testing

Reply:
done